packages/debug/src/browser/view/debug-toolbar-widget.tsx

```diff
@@ -77,6 +81,36 @@ export class DebugToolBar extends ReactWidget {
             {this.renderStart()}
         </React.Fragment>;
     }
+
+    protected renderContributedCommands(): React.ReactNode {
+        const match = ({ when }: MenuNodeMetadata) => !when || this.contextKeyService.match(when);
+        return this.menuModelRegistry
+            .getMenu(DebugToolBar.MENU).children
+            .filter(CompoundMenuNode.is)
+            .filter(compoundNode => match(compoundNode))
+            .map(compoundNode => compoundNode.children)
+            .reduce((acc, curr) => acc.concat(curr), [])
+            .filter(CommandMenuNode.is)
+            .filter(commandNode => match(commandNode))
+            .map(commandNode => this.debugAction(commandNode));
+    }
+
+    protected debugAction(node: CommandMenuNode): React.ReactNode {
```

Comment:
`toDebugAction` or `createDebugAction` would more accurately describe this method.

Reply:
https://github.com/eclipse-theia/theia/pull/11984/commits/2b6e114695481f0b723230fbb93f03eba3381ecb